src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php

```diff
@@ -1745,19 +1743,41 @@ private function registerMessengerConfiguration(array $config, ContainerBuilder
 
         $messageToSendersMapping = [];
         $messagesToSendAndHandle = [];
+        if ($config['failure_transport']) {
+            $failureTransport = $config['failure_transport'];
+
+            $loader->load('messenger_failure.xml');
+            $container->getDefinition('console.command.messenger_failed_messages_retry')
+                ->replaceArgument(0, $failureTransport)
+                ->replaceArgument(4, $transportRetryReferences[$failureTransport] ?? null);
+            $container->getDefinition('console.command.messenger_failed_messages_show')
+                ->replaceArgument(0, $failureTransport);
+            $container->getDefinition('console.command.messenger_failed_messages_remove')
+                ->replaceArgument(0, $failureTransport);
+
+            // push routing for FailedMessage to the failure transport
+            if (!isset($messageToSendersMapping[FailedMessage::class])) {
+                $messageToSendersMapping[FailedMessage::class] = [];
+            }
+            $messageToSendersMapping[FailedMessage::class][] = $failureTransport;
+
+            // in case this is a tagged sender, make sure it's in the aliases
+            if (!isset($senderAliases[$failureTransport])) {
+                $senderAliases[$failureTransport] = $failureTransport;
+            }
+        }
```

Comment:
This section was mostly just moved from below. The `$messagesToSendersMapping` is the new part: it adds `routing` for the `FailedMessage` class to whatever the failure transport is.